GameSentenceMiner/web/database_api.py

```diff
@@ -780,4 +782,165 @@ def api_stats():
+                return jsonify({'error': 'No file selected'}), 400
+            
+            # Validate file type
+            if not file.filename.lower().endswith('.csv'):
+                return jsonify({'error': 'File must be a CSV file'}), 400
+            
+            # Read and parse CSV
+            try:
+                # Read file content as text with proper encoding handling
+                file_content = file.read().decode('utf-8-sig')  # Handle BOM if present
+                
+                # First, get the header line manually to avoid issues with multi-line content
+                lines = file_content.split('\n')
+                if len(lines) == 1 and not lines[0].strip():
+                    return jsonify({'error': 'Empty CSV file'}), 400
+                
+                header_line = lines[0].strip()
+                logger.info(f"Header line: {header_line}")
+                
+                # Parse headers manually
+                header_reader = csv.reader([header_line])
+                try:
+                    headers = next(header_reader)
+                    headers = [h.strip() for h in headers]  # Clean whitespace
+                    logger.info(f"Parsed headers: {headers}")
+                except StopIteration:
+                    return jsonify({'error': 'Could not parse CSV headers'}), 400
+                
+                # Validate headers
+                expected_headers = {'uuid', 'given_identifier', 'name', 'line', 'time'}
+                actual_headers = set(headers)
+                
+                if not expected_headers.issubset(actual_headers):
+                    missing_headers = expected_headers - actual_headers
+                    # Check if this looks like a stats CSV instead of lines CSV
+                    if 'client' in actual_headers and 'chars_read' in actual_headers:
+                        return jsonify({
+                            'error': 'This appears to be an ExStatic stats CSV. Please upload the ExStatic lines CSV file instead. The lines CSV should contain columns: uuid, given_identifier, name, line, time'
+                        }), 400
+                    else:
+                        return jsonify({
+                            'error': f'Invalid CSV format. Missing required columns: {", ".join(missing_headers)}. Expected format: uuid, given_identifier, name, line, time. Found headers: {", ".join(actual_headers)}'
+                        }), 400
+                
+                # Now parse the full CSV with proper handling for multi-line fields
+                file_io = io.StringIO(file_content)
+                csv_reader = csv.DictReader(file_io, quoting=csv.QUOTE_MINIMAL, skipinitialspace=True)
+                
+                # Process CSV rows
+                imported_lines = []
+                games_set = set()
+                errors = []
+                seen_uuids = set()  # Track UUIDs within this import batch
```

Comment:
ExStatic assigns UUIDs to all lines, this is so we don't import the same line twice